pom.xml

```diff
@@ -33,154 +33,397 @@
-      <groupId>org.apache.hive</groupId>
-      <artifactId>hive-service</artifactId>
-      <version>${hive.version}</version>
-    </dependency>
-
-    <dependency>
-      <artifactId>tez-dag</artifactId>
-      <groupId>org.apache.tez</groupId>
-      <version>${tez.version}</version>
-    </dependency>
-
-    <dependency>
-      <artifactId>tez-common</artifactId>
-      <groupId>org.apache.tez</groupId>
-      <version>${tez.version}</version>
-    </dependency>
-
-    <dependency>
-      <artifactId>tez-mapreduce</artifactId>
-      <groupId>org.apache.tez</groupId>
-      <version>${tez.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hive</groupId>
-      <artifactId>hive-contrib</artifactId>
-      <version>${hive.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.reflections</groupId>
-      <artifactId>reflections</artifactId>
-      <version>0.9.8</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-mapreduce-client-common</artifactId>
-      <version>${hadoop.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-mapreduce-client-core</artifactId>
-      <version>${hadoop.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>com.github.stefanbirkner</groupId>
-      <artifactId>system-rules</artifactId>
-      <version>1.19.0</version>
-      <scope>test</scope>
-    </dependency>
-
-    <!-- Always put this before JUnit or the class loader might load the
-      wrong Matcher -->
-    <dependency>
-      <groupId>org.hamcrest</groupId>
-      <artifactId>hamcrest-all</artifactId>
-      <version>1.3</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
-      <version>${mockito.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-junit-jupiter</artifactId>
-      <version>${mockito.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>4.13.2</version>
-      <scope>provided</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <version>${junit.jupiter.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.junit.vintage</groupId>
-      <artifactId>junit-vintage-engine</artifactId>
-      <version>${junit.jupiter.version}</version>
-    </dependency>
-
-  </dependencies>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <hadoop.version>3.4.0</hadoop.version>
+        <hive.version>4.0.1</hive.version>
```

Comment:
We're going to have to think about how we handle this as it's obviously a backwards-incompatible change. What we did in the past with `hive-3` was we put that into a separate branch and kept `main` for hive 2. At some point we decided Hive 3 had reached critical mass and we then swapped them around. So we'll probably want to do something similar here, we'll have a think. For now we can continue looking at the PR against `main` but we might want to move this over to a different branch.

Reply:
Sure, thanks. I would also like to fix here some other issues as well. But lets focus on this PR first.